### PR TITLE
Implement SamplingLogger

### PR DESCRIPTION
**Pull request by gpang**
This is a backport of the same feature from the master branch. This sampling logger helps with spammy log messages.

### CODE REVIEW

**Comment by gpang** [comment]
@calvinjia please take a look

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -3678,8 +3680,8 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
         try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
           UnderFileSystem ufs = ufsResource.get();
           if (ufs.isObjectStorage()) {
-            LOG.debug("setOwner/setMode is not supported to object storage UFS via Alluxio. "
-                + "UFS: " + ufsUri + ". This has no effect on the underlying object.");
+            SAMPLING_LOG.debug("setOwner/setMode is not supported to object storage UFS via "
+                + "Alluxio. UFS: {}. This has no effect on the underlying object.", ufsUri);
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: Will this help in the case `ufsUri` is different? Also this is not part of the original change so maybe we want to separate this out (or is the sampling logger change not cherry-pickable anyway)?

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
@calvinjia updated to just include the logger change.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**